packages/emmet/index.ts

```diff
@@ -1,106 +1,210 @@
+					validateLocation = true;
+					rootNode = getRootNode(document, true);
+					if (!rootNode) {
+						return;
+					}
+					const flatNode = getFlatNode(rootNode, offset, true);
+					const embeddedCssNode = getEmbeddedCssNodeIfAny(document, flatNode, position);
+					currentNode = getFlatNode(embeddedCssNode, offset, true);
+				}
 
-					return emmet.doComplete(textDocument, position, syntax, emmetConfig);
-				},
-			};
+				if (validateLocation && !await isValidLocationForEmmetAbbreviation(context, document, rootNode, currentNode, syntax, offset, extractAbbreviationResults.abbreviationRange)) {
+					return;
+				}
 
-			async function getEmmetConfig(syntax: string): Promise<emmet.VSCodeEmmetConfig> {
-
-				const emmetConfig: any = await context.env.getConfiguration?.<emmet.VSCodeEmmetConfig>('emmet') ?? {};
-				const syntaxProfiles = Object.assign({}, emmetConfig['syntaxProfiles'] || {});
-				const preferences = Object.assign({}, emmetConfig['preferences'] || {});
-
-				// jsx, xml and xsl syntaxes need to have self closing tags unless otherwise configured by user
-				if (syntax === 'jsx' || syntax === 'xml' || syntax === 'xsl') {
-					syntaxProfiles[syntax] = syntaxProfiles[syntax] || {};
-					if (typeof syntaxProfiles[syntax] === 'object'
-						&& !syntaxProfiles[syntax].hasOwnProperty('self_closing_tag') // Old Emmet format
-						&& !syntaxProfiles[syntax].hasOwnProperty('selfClosingStyle') // Emmet 2.0 format
-					) {
-						syntaxProfiles[syntax] = {
-							...syntaxProfiles[syntax],
-							selfClosingStyle: 'xml'
-						};
+				let isNoisePromise: Thenable<boolean> = Promise.resolve(false);
+
+				// Fix for https://github.com/microsoft/vscode/issues/32647
+				// Check for document symbols in js/ts/jsx/tsx and avoid triggering emmet for abbreviations of the form symbolName.sometext
+				// Presence of > or * or + in the abbreviation denotes valid abbreviation that should trigger emmet
+				if (!isStyleSheet(syntax) && (document.languageId === 'javascript' || document.languageId === 'javascriptreact' || document.languageId === 'typescript' || document.languageId === 'typescriptreact')) {
+					const abbreviation: string = extractAbbreviationResults.abbreviation;
+					// For the second condition, we don't want abbreviations that have [] characters but not ='s in them to expand
+					// In turn, users must explicitly expand abbreviations of the form Component[attr1 attr2], but it means we don't try to expand a[i].
+					if (abbreviation.startsWith('this.') || /\[[^\]=]*\]/.test(abbreviation)) {
+						isNoisePromise = Promise.resolve(true);
+					} else {
+						// TODO
+						// isNoisePromise = vscode.commands.executeCommand<vscode.SymbolInformation[] | undefined>('vscode.executeDocumentSymbolProvider', document.uri).then(symbols => {
```

Comment:
This is currently not implemented and requires @volar/language-service to expose a API to get symbol informations.